app/models/spam_domain.rb

```diff
@@ -20,17 +20,15 @@ def fix_asn_tags
     asn_query = `dig +short "$(dig +short '#{domain.tr("'", '')}' | awk -F. '/[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+/ {print $4"."$3"." $2"."$1;exit}').origin.asn.cymru.com" TXT` # rubocop:disable Metrics/LineLength
     asn = asn_query.strip.tr('"', '').split('|')[0]&.strip
     return unless asn.present?
-    prev_domain_tags = domain_tags.where(special: true).select(:name).map(&:name).select { |dt| dt.start_with?('AS-') }
+    domain_tags.where(special: true).select(:name).map(&:name).select { |dt| dt.start_with?('AS-') }.each do |dtn|
+      domain_tags.delete(DomainTag.find_by(name: dtn))
+    end
     asn.split.each do |as|
       desc = `dig +short AS#{as}.asn.cymru.com TXT`.strip.tr('"', '').split('|')[-1]&.strip
       tag = DomainTag.find_or_create_by(name: "AS-#{as}", special: true)
-      prev_domain_tags -= ["AS-#{as}"]
       tag.update(description: "Domains under the Autonomous System Number #{as} - #{desc}.") unless tag.description.present?
       domain_tags << tag
```

Comment:
I'd prefer to fix this by sticking an `unless domain_tags.include? tag` on this line.

Reply:
It's just going to be slightly better than clearing all the AS tags then re-creating them.